cloud/amazon/iam_policy.py

```diff
@@ -183,6 +183,14 @@ def role_action(module, iam, name, policy_name, skip, pdoc, state):
     current_policies = [cp for cp in iam.list_role_policies(name).
                                         list_role_policies_result.
                                         policy_names]
+  except boto.exception.BotoServerError as e:
+    if e.error_code == "NoSuchEntity":
+      # Role doesn't exist so it's safe to assume the policy doesn't either
```

Comment:
What do you think about doing a check for the policy here? Isn't it possible that the policy exists even if the role doesn't?